### PR TITLE
11878 allow date question searchability

### PR DESCRIPTION
**Pull request by cooperka**
(See new docs in code changes for details.)

This was tricky and interesting! I did not know what sql "tsvector" was until now.

It's managed by [hair_trigger](https://github.com/jenseng/hair_trigger).

### CODE REVIEW

**Comment by cooperka** [comment]
Specs are flappers. I also realized this may require a migration for legacy submissions; I'll test this case when I return, but it's not blocking code review.

File: app/models/results/answer_search_vector_updater.rb [hunk 1]
```diff
@@ -20,6 +23,9 @@ def trigger_expression
       <<-SQL.squish
         TO_TSVECTOR('simple', COALESCE(
           new.value,
+          to_char(new.date_value, 'YYYY-MM-DD'),
+          to_char(new.time_value, 'HH24hMImSSs'),
+          to_char(new.datetime_value, 'YYYY-MM-DD HH24hMImSSs'),
```
Comment (jpdimond) [inline app/models/results/answer_search_vector_updater.rb:28]: good to know where to add columns for search!

**Review by jpdimond — APPROVED** [review]
LGTM! are you wanting me to manually test?

**Comment by cooperka** [comment]
I've tested pretty extensively, no need unless you feel inclined to. Thanks!

**Pull request merged**